src/hotspot/share/gc/g1/g1_globals.hpp

```diff
@@ -377,6 +377,11 @@
           "retained region restore purposes.")                              \
           range(1, 256)                                                     \
                                                                             \
+  product(uint, G1NumCardsCostSampleThreshold, 1000, DIAGNOSTIC,            \
```

Comment:
Unfounded hunch: an inclusive threshold here is probably easier to reason.

Reply:
Either is fine with me. Changed.